C Error in handling .split().pop() expression

### Describe the bug

I got an error while attempting to chain the .split('.').pop() expression to obtain the last item after '.' in the string (ex: file extension). The error occurs during the compilation process, and the compiler output indicates a problem with the generated C code. Upon closer inspection its clear that the C code is syntactically incorrect.

### Reproduction Steps

1. Use the following code snippet:
    ```v
    module main
    fn main() {
        mut path := 'hello/file.txt'
        extension := path.split('.').pop()
        println(extension)
    }
    ```
2. Run the code using `v run file.v`

### Expected Behavior

The compiler should correctly handle the .split().pop() expression, allowing the program to run without errors.

### Current Behavior

The compiler generates C code that leads to errors during compilation, as indicated in the provided compiler output. It generates syntactically incorrect C code.

**Generated error message:**
```
vishw@waifustation C:\Projects\Temp\vlang $ v run .\main.v
==================
C:/U....78.tmp.c:6822: warning: implicit declaration of function 'tcc_backtrace'
C:/U....78.tmp.c:12811: error: ')' expected (got ";")
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.
```

**Generated C code at line 12811:**
```c
string extension = (*(string*)array_pop(ADDR(Array_string, string_split(path, _SLIT("."))));
```

Which is clearly mission and ending `)`

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.3 e19b2dd

### Environment details (OS name and version, etc.)

V full version: V 0.4.3 4b31092.772f114
OS: windows, Microsoft Windows 11 Home v22621 64-bit
Processor: 12 cpus, 64bit, little endian,

getwd: C:\Projects\Temp\vlanger
vexe: C:\Program Files\v\v.exe
vexe mtime: 2023-12-07 13:03:24

vroot: contains spaces, value: C:\Program Files\v
VMODULES: OK, value: C:\Users\vishw\.vmodules
VTMP: OK, value: C:\Users\vishw\AppData\Local\Temp\v_0

Git version: git version 2.41.0.windows.1
Git vroot status: weekly.2023.48-62-g772f1149 (196 commit(s) behind V master)
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,
operable program or batch file.

thirdparty/tcc status: thirdparty-windows-amd64 a39eb79b

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -1372,7 +1372,7 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 		}
 		g.get_free_method(rec_type)
 	}
-	mut has_cast := false
+	mut cast_n := 0
 
 	receiver_type_name = g.resolve_receiver_name(node, unwrapped_rec_type, final_left_sym,
 		left_sym, typ_sym)
@@ -1380,7 +1380,7 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 		&& node.name in ['clear', 'repeat', 'sort_with_compare', 'sorted_with_compare', 'free', 'push_many', 'trim', 'first', 'last', 'pop', 'clone', 'reverse', 'slice', 'pointers'] {
 		if node.name in ['last', 'first', 'pop'] {
 			return_type_str := g.typ(node.return_type)
-			has_cast = true
+			cast_n++
 			g.write('(*(${return_type_str}*)')
 		}
 	}
@@ -1485,7 +1485,7 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 		if !is_range_slice {
 			if !node.left.is_lvalue() {
 				g.write('ADDR(${rec_cc_type}, ')
-				has_cast = true
+				cast_n++
 			} else if !is_array_method_first_last_repeat && !(left_type.has_flag(.shared_f)
 				&& left_type == node.receiver_type) {
 				g.write('&')
@@ -1592,8 +1592,8 @@ fn (mut g Gen) method_call(node ast.CallExpr) {
 			g.write('->val')
 		}
 	}
-	if has_cast {
-		g.write(')')
+	if cast_n > 0 {
+		g.write(')'.repeat(cast_n))
 	}
 	is_variadic := node.expected_arg_types.len > 0
 		&& node.expected_arg_types.last().has_flag(.variadic)

--- a/vlib/v/tests/method_call_chain_test.v
+++ b/vlib/v/tests/method_call_chain_test.v
@@ -1,0 +1,7 @@
+// for issue 20395
+// Phenomenon: cgen brackets are asymmetric in chained method calls.
+fn test_main() {
+	mut path := 'hello/file.txt'
+	extension := path.split('.').pop()
+	assert extension == 'txt'
+}

